### PR TITLE
Nv 3004 the new push editor

### PR DESCRIPTION
**Pull request by BiswaViraj**
### What change does this PR introduce?
- implemented new editor screen with preview component

### Other information (Screenshots)

https://github.com/novuhq/novu/assets/29251776/4ffd01f8-adbf-4b60-885b-baf55e9c4402

### CODE REVIEW

File: apps/web/src/pages/templates/hooks/usePreviewPushTemplate.ts [hunk 1]
```diff
@@ -47,11 +44,11 @@ export const usePreviewPushTemplate = (locale?: string) => {
   useEffect(() => {
     getPushPreview({
       locale,
-      content: previewData.current.templateContent,
-      payload: previewData.current.processedVariables,
-      title: previewData.current.templateTitle,
+      content: templateContent,
+      payload: processedVariables,
```
Comment (ainouzgali) [inline apps/web/src/pages/templates/hooks/usePreviewPushTemplate.ts:48]: In the getPushPreview implementation(in the mutateAsync call) - you need to make sure to parse the payload : `payload: JSON.parse(payload)`
as processedVariables is a string

**Review by ainouzgali — COMMENTED** [review]
Great work @BiswaViraj !! 
Left you a few small comments :)
Another note - in the preview that is shown with the editor , the edit on hover should not be shown 
Screen Shot 2024-02-15 at 13 09 42

File: apps/web/src/pages/templates/hooks/usePreviewPushTemplate.ts [hunk 2]
```diff
@@ -45,13 +42,15 @@ export const usePreviewPushTemplate = (locale?: string) => {
...
-      content: previewData.current.templateContent,
-      payload: previewData.current.processedVariables,
-      title: previewData.current.templateTitle,
-    });
-  }, [getPushPreview, locale, previewData]);
+    if (!disabled) {
+      getPushPreview({
+        locale,
+        content: templateContent,
+        payload: JSON.parse(processedVariables),
```
Comment (ainouzgali) [inline apps/web/src/pages/templates/hooks/usePreviewPushTemplate.ts:49]: Sorry, I meant that here it should stay the same (`PayloadType.payload`  is type string)
```suggestion
        payload: processedVariables,
```
But in the implementation it should parse it  
Screen Shot 2024-02-15 at 14 17 43

Comment (BiswaViraj) [reply apps/web/src/pages/templates/hooks/usePreviewPushTemplate.ts:49]: Oh ok, my bad.
Updating it

File: apps/web/src/pages/templates/components/TemplatePushEditor.tsx [hunk 3]
```diff
@@ -27,42 +33,61 @@ export function TemplatePushEditor() {
...
-            {...field}
-            data-test-id="pushNotificationContent"
-            error={stepFormErrors ? stepFormErrors.template?.content?.message : undefined}
-            disabled={readonly}
-            minRows={4}
-            value={field.value || ''}
-            label="Push message content"
-            placeholder="Add notification content here..."
+
+      <Grid>
```
Comment (ainouzgali) [inline apps/web/src/pages/templates/components/TemplatePushEditor.tsx:37]:
```suggestion
      <Grid gutter={24}>
```
from the design

**Review by ainouzgali — APPROVED** [review]

**Pull request merged**